### PR TITLE
feat: script to update sdk version, additional husky rule

### PR DESCRIPTION
**Pull request by scastleman-immutable**
# Summary
New script to update the SDK version 

# Detail and impact of the change

## Added 
Script in sdk-sample-app to update __SDK_VERSION__ to latest git tag, updated tests to ensure this version does not get committed. Added husky hook to test project to prevent projects with failing tests being committed as further safeguard.

# Anything else worth calling out?
Needs pipeline change before deployed sample app will function.

### CODE REVIEW

File: .github/workflows/passport-sdk-sample-app-deployment.yaml [hunk 1]
```diff
@@ -46,6 +46,9 @@ jobs:
       - name: Install dependencies
         run: yarn install --immutable
 
+      - name: Ppdate SDK version
+        run: yarn workspace @imtbl/passport-sdk-sample-app update-sdk-version
```
Comment (yundifu) [inline .github/workflows/passport-sdk-sample-app-deployment.yaml:50]: does this work on CI now ?

**Review by yundifu — APPROVED** [review]

**Review by carmen0208 — APPROVED** [review]

**Comment by scastleman-immutable** [comment]
> LGTM, but this seems quite specific for the passport package. I wonder whether other packages will run into this issue. Was this the proposed approach from the SDK team?

Yes, Carmen spoke with Nik Ho and he recommended this approach.

File: .husky/pre-commit [hunk 2]
```diff
@@ -2,3 +2,4 @@
 . "$(dirname -- "$0")/_/husky.sh"
 
 yarn lint-staged
+yarn test --watchAll=false
```
Comment (cajames) [inline .husky/pre-commit:5]: Why was this added?
